test/odbc/test_pg_datatype_float.cpp

```diff
@@ -0,0 +1,610 @@
+const int COL_LENGTH[NUM_COLS]={10,308,308};
+const int COL_SCALE[NUM_COLS] = {0, 308, 308};
+
+// unsure of how these values are calculated, but they match with postgres' (non-sys) decimal type
+// Postgres documentation just says they're bytes size are variable
+
+const string COL_TYPES[NUM_COLS] = {
+  DATATYPE ,
+  DATATYPE ,
+  DATATYPE 
+};
+
+vector<pair<string, string>> TABLE_COLUMNS = {
+  {COL_NAMES[0], COL_TYPES[0] + " PRIMARY KEY"},
+  {COL_NAMES[1], COL_TYPES[1]},
+  {COL_NAMES[2], COL_TYPES[2]}
+};  
+double StringToDouble(const string &value){
+    return atof(value.c_str());
+  }
+
+const string float_15 = "1e308";
+// const string float_383 = "0.12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123";
+const string float_383 ="-1e307";
+class PSQL_DataTypes_Float : public testing::Test{
+
+  void SetUp() override {
+
+    OdbcHandler test_setup;
+    test_setup.ConnectAndExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+  
+
+  void TearDown() override {
+
+    OdbcHandler test_cleanup;
+    test_cleanup.ConnectAndExecQuery(DropObjectStatement("VIEW", VIEW_NAME));
+    test_cleanup.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+  }
+};
+
+// pads 0 at the end of the results depending on the scale number
+// double FormatDecWithScale(string decimal, const int &scale) {
+
+//   size_t dec_pos = decimal.find('.');
+
+//   if (dec_pos == std::string::npos) {
+//     if (scale == 0) // if no decimal sign and scale is 0, no need to append
+//       return atof(decimal);
+//     dec_pos = decimal.size();
+//     decimal += ".";
+//   }
+
+//   // add extra 0s
+//   int zeros_needed = scale - (decimal.size() - dec_pos - 1);
+//   for (int i = 0; i < zeros_needed; i++) {
+//     decimal += "0";
+//   }
+
+//   return atof(decimal);
+// }
+
+// helper function to initialize insert string (1, "", "", ""), etc.
+string InitializeInsertString(const vector<vector<string>> &inserted_values) {
+
+  string insert_string{};
+  string comma{};
+
+  for (int i = 0; i< inserted_values.size(); ++i) {
+
+    insert_string += comma + "(";
+    string comma2{};
+
+    for (int j = 0; j < NUM_COLS; j++) {
+      insert_string += comma2 + inserted_values[i][j];
+      comma2 = ",";
+    }
+
+    insert_string += ")";
+    comma = ",";
+  }
+  return insert_string;
+}
+
+TEST_F(PSQL_DataTypes_Float, ColAttributes) {
+
+  const string NAME_EXPECTED = "float8";
+  const string DISPLAY_SIZE_EXPECTED = "idk";
+  const int Lenghth_expected[NUM_COLS]={8,17,17};
+  const int PREICSION_EXPECTED=0;
+  const int SCALE_EXPECTED=0;
+  const int COL_LENGTH_EXPECTED[NUM_COLS]={24,24,24};
+  const int BUFFER_SIZE = 256;
```

Comment:
Probably move this as a global, it'll be useful for other tests e.g. checking composite keys using `SQLPrimaryKeys(...)`.